### PR TITLE
Focus goes to next EditText after retrieving contact

### PR DESCRIPTION
**Pull request by rohands**
Fix for #218

### CODE REVIEW

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java [hunk 1]
```diff
@@ -43,21 +44,34 @@
...
 
-        Button okButton = (Button) findViewById(R.id.okButton);
+        comradeEditText.add(comrade1EditText);
+        comradeEditText.add(comrade2EditText);
+        comradeEditText.add(comrade3EditText);
+        comradeEditText.add(comrade4EditText);
+        comradeEditText.add(comrade5EditText);
+        comradeEditText.add(comrade6EditText);
+
+
```
Comment (sandarumk) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: One blank line is enough

File: app/src/main/res/layout/activity_trustees.xml [hunk 2]
```diff
@@ -236,7 +242,7 @@
             android:text="@string/add_comrades_save"
             android:layout_below="@id/comrade6EditText"
             android:layout_marginTop="@dimen/trustee_margin_top"
-            android:layout_marginBottom="@dimen/trustee_margin_bottom"/>
+            android:layout_marginBottom="@dimen/trustee_margin_bottom" />
```
Comment (sandarumk) [inline app/src/main/res/layout/activity_trustees.xml]: The space at the end is not exactly needed.

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java [hunk 3]
```diff
@@ -43,21 +44,34 @@
...
     SharedPreferences.Editor editor;
+    private Button okButton;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_trustees);
 
-        System.out.println(comradeEditText.size());
-        comradeEditText.add((EditText) findViewById(R.id.comrade1EditText));
```
Comment (sandarumk) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: Explain the rationale behind your choice of removing these and breaking the functionality in to two.

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java [hunk 4]
```diff
@@ -305,5 +323,43 @@ private boolean noDuplicateNumber() {
         }
         return noDuplicate;
     }
+
+    /**
+     * Set focus to the next EditText after picking contact
+     */
+    public void setFocusOnNextView() {
```
Comment (sandarumk) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: For finding the index of the selected button try using tag of EditText (phoneInput) instead of introducing new tags for the buttons.

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java (hunk 3 above)
Comment (rohands) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: I had initially planned to get hold of the views individually so that I can
request focus on the view next to the currently focused EditText. I found
that there was a reference being made to the EditTexts but they were only
made to add them to the list which is not so useful since I can't make use
of it elsewhere if I'm trying to make any changes to the view in future.
If we think from the optimisation point of view, the newly written code is
more efficient than the earlier written code in terms of reusablility.

Comment (rohands) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: @sandarumk
I'm sorry about the extra blanks and white spaces. I didn't happen to
notice them. I shall make all changes into the next commit so please let me
know if I can continue

Comment (sandarumk) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: If it is necessary to make changes in the future, iterating list is easy and consistent than making individual changes. In optimization point of view the previous code is much better than the newer one when considering the overhead of creating local variable and adding it to the list happen in two executions

**Comment by sandarumk** [comment]
@rohands you can make a fresh PR with the suggestions and corrections.

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java (hunk 4 above)
Comment (rohands) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: Tags for buttons were already added by the developer who added the contact retrieval feature. I am not trying to find the index of the selected button. I am trying to find the index of the next view which doesn't have any numbers registered.

Comment (rohands) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: Please take a look at the existing code once.

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java (hunk 3 above)
Comment (rohands) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: I think a temporary variable is created either way. Creating a temporary variable and adding it to the list also happens on two clock cycles.
However, as you mentioned, I feel this is not required currently. I shall undo the changes

**Comment by sandarumk** [comment]
Since you mentioned the CPU cycles, this is what really happens. There is an extra store operation with the creation of the local variable.
[Uploading extra cpu cylce.txt…]()

**Comment by sandarumk** [comment]
[extra cpu cylce.txt](https://github.com/systers/PCSA-Android/files/221286/extra.cpu.cylce.txt)
Since you are interested in CPU cycles, this is what really happens. If you can see, there will be an extra store instruction due to the creation of the local variable.

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java [hunk 5]
```diff
@@ -305,5 +323,43 @@ private boolean noDuplicateNumber() {
...
+
+        if(index >= 0 && index < comradeEditText.size())
+        {
+            EditText nextFocus = comradeEditText.get(index);
+            if(nextFocus.getText().toString().equals(""))
+                nextFocus.requestFocus();
+            else
+                okButton.requestFocus();
+        }
+        else
```
Comment (sandarumk) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: Can you explain a scenario where this else is called?

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java (hunk 4 above)
Comment (sandarumk) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: Sorry if I confused you What I meant to tell is that the phoneInput is the selected button. The previous developer has done the same thing when creating the phoneInput, you can use that without doing the same thing again. Please go through the current implementation and try to understand what is done there. Then you will be able to implement this more easily.

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java (hunk 5 above)
Comment (rohands) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: EditText 1 : filled
EditText 2 : filled
EditText 3 : unfilled
EditText 4 : filled
EditText 5 : filled
EditText 6 : FOCUS
After filling EditText 6, control goes to EditText 3

File: app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java (hunk 4 above)
Comment (rohands) [inline app/src/main/java/com/peacecorps/pcsa/circle_of_trust/Trustees.java]: Oh okay, I got it. Thank you
